www/NativePageTransitions.js

```diff
@@ -126,21 +126,26 @@ NativePageTransitions.prototype._validateHref = function (href, errCallback) {
   if (hrf.indexOf('www/') == -1) {
     console.log('Probably running inside a companion app, your app may crash if your html file is not in the root!');
     // hrf is something like file:///data/.../index.html
-    currentHref = hrf.substr(hrf.lastIndexOf('/')+1);
+    //we need the path after the hash for the remote views (for example /index.html#a/b/page1.html)
+    currentHref = hrf.substr(hrf.lastIndexOf('/index.html') + 1);
   } else {
     currentHref = hrf.substr(hrf.indexOf('www/')+4);
   }
   // if no href was passed the transition should always kick in
   if (href) {
     // if only hash nav, do it in JS for Android
-    // (I'm a little reluctant to depend on 'device' only for this: device.platform == "Android")
-    if (href.indexOf('#') == 0 && navigator.userAgent.indexOf("Android") > -1) {
+    // (navigator.userAgent gives incorrect data sometimes, so it's more robust to check for device.platform)
+    if (href.indexOf('#') == 0 && device.platform == "Android") {
```

Comment:
I understand, I meant to say: could you please add a dependency to `plugin.xml` for `cordova-plugin-device`? Otherwise it may break a build here and there after merging this PR.